lib/segment/src/index/hnsw_index/hnsw.rs

```diff
@@ -94,8 +94,13 @@ impl<TGraphLinks: GraphLinks> HNSWIndex<TGraphLinks> {
         } else {
             let vector_storage = vector_storage.borrow();
             let available_vectors = vector_storage.available_vector_count();
-            let full_scan_threshold = hnsw_config.full_scan_threshold.saturating_mul(BYTES_IN_KB)
-                / (vector_storage.vector_dim() * VECTOR_ELEMENT_SIZE);
+            let full_scan_threshold = if available_vectors > 0 {
```

Comment:
Discussion point. What to do if storage is empty and is this estimation correct? It affects HNSW search